subsys/logging/log_backend_swo.c

```diff
@@ -144,10 +152,11 @@ static void log_backend_swo_sync_hexdump(
 }
 
 const struct log_backend_api log_backend_swo_api = {
-	.put = IS_ENABLED(CONFIG_LOG_IMMEDIATE) ? NULL : log_backend_swo_put,
-	.put_sync_string = IS_ENABLED(CONFIG_LOG_IMMEDIATE) ?
+	.process = IS_ENABLED(CONFIG_LOG2) ? log_backend_swo_process : NULL,
+	.put = IS_ENABLED(CONFIG_LOG_MODE_DEFERRED) ? log_backend_swo_put : NULL,
+	.put_sync_string = IS_ENABLED(CONFIG_LOG_MODE_IMMEDIATE) ?
 			log_backend_swo_sync_string : NULL,
-	.put_sync_hexdump = IS_ENABLED(CONFIG_LOG_IMMEDIATE) ?
+	.put_sync_hexdump = IS_ENABLED(CONFIG_LOG_MODE_IMMEDIATE) ?
```

Comment:
This and updates to `.put`, `.put_sync_string` look like a bugfix, an unrelated change. This should likely go into a separate commit.

Reply:
it's not a bugfix, without v2 support using `IS_ENABLED(CONFIG_LOG_IMMEDIATE)` was ok but `CONFIG_LOG_IMMEDIATE` is set for both immediate versions so now we need better distinction and thus `CONFIG_LOG_MODE_IMMEDIATE` is used which is v1 immediate mode (with `CONFIG_LOG2_MODE_IMMEDIATE` for v2)